app/types/types.ts

```diff
@@ -41,6 +41,17 @@ export interface ColDataCardDef<T extends Record<string, unknown>> {
     column: ColDataCardDef<T>
   ) => string;
   valueRenderer?: (record: T, column: ColDataCardDef<T>) => ReactNode;
+  /**
+   * Function that returns a custom component, used for display and/or editing
+   * when the column is dependent on other columns of the same record
+   */
+  customComponent?: (
+    record: T,
+    onCancel: () => void,
+    mutation: string,
+    onSaveCallback: () => Promise<void>,
+    isEditingThisColumn: boolean,
+  ) => ReactNode;
```

Comment:
This is great ✨  I _think_ we want this property to be title cased like a react component, i.e. `CustomComponent` ?